### PR TITLE
Single char storage

### PR DESCRIPTION
**Pull request by rogermc2**

### CODE REVIEW

File: src/gl/implementation/gl-objects.adb [hunk 1]
```diff
@@ -43,17 +43,7 @@ package body GL.Objects is
...
-               exception
-                  when others =>
-                     --  cannot let this escape as we're in a Finalize call and
-                     --  thus that error cannot be properly catched. Chances are
-                     --  that if the destructor fails, the context already has
-                     --  vanished and thus we do not need to worry about
-                     --  anything.
-                     null;
-               end;
+               Reference.Destructor (Reference);
```
Comment (flyx) [inline src/gl/implementation/gl-objects.adb]: I assume this change has been accidental, please revert it. It reverts an earlier fix by myself.

File: examples/freetype_gl/single_storage/src/texture_manager.ads [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+   subtype V_Buffer is GL.Objects.Buffers.Buffer;
+   --  2D quad as two triangles requires 2 * 3 vertices of 4 floats
+   subtype Vertex_Array is GL.Types.Singles.Vector4_Array (1 .. 6);
+
+   procedure Setup_Graphic (Vertex_Buffer : in out V_Buffer;
+                            aTexture      : in out GL.Objects.Textures.Texture;
+                            X, Y          : GL.Types.Single;
+                            Scale         : GL.Types.Single;
+                            Char          : Character := 'g');
+end Texture_Manager;
```
Comment (flyx) [inline examples/freetype_gl/single_storage/src/texture_manager.ads]: Is this shareable with your previous PR? could be in `examples/freetype_gl/common`.

File: src/gl/implementation/gl-objects.adb (hunk 1 above)
Comment (rogermc2) [reply src/gl/implementation/gl-objects.adb]: Definitely accidental!
Not sure how to find the offending commit.
I'll reinstate from master.

File: examples/freetype_gl/single_storage/src/texture_manager.ads (hunk 2 above)
Comment (rogermc2) [reply examples/freetype_gl/single_storage/src/texture_manager.ads]: I'll look into this.

Comment (rogermc2) [reply examples/freetype_gl/single_storage/src/texture_manager.ads]: Something seems to have gone drastically wrong.
This code seems to be just a copy of the previous PR.
I'll have to see if I can recover the real code tomorrow.
If I can't fix it, I'll have to withdraw this PR and move on to the string example.

Comment (rogermc2) [reply examples/freetype_gl/single_storage/src/texture_manager.ads]: I think I've fixed it, thanks to git revert.

**Comment by rogermc2** [comment]
Oh, lots of conflicts!
I'll have to attend to them tomorrow.
Advice will be appreciated.

**Comment by rogermc2** [comment]
I'll close it while I sort out the mess I've made.

**Pull request closed without merging**